### DISCUSSION TITLE
Profiling async functions out of the statemachine formatting

### DISCUSSION THREAD

**Discussion by StanLipman** [dotnet/roslyn#75078] [Q&A] [answered]
Can I be sure that every async function that gets split into pieces by the statemachine has the same format? 
As of now I found that every async function ends with <func_Name>d__{corresponding func_ID} 
I also found out that the funcID is based on the order in the corresponding file, the first function gets 0, second 1, etc.

![image](https://github.com/user-attachments/assets/795b27d5-c8fc-4639-a499-6bbd6d691ecf)

**Comment by 333fred** [comment]
What's your actual question?

**Reply by StanLipman** [reply]
Can i be sure that every async function that gets put into the statemachine has the same format? (<{func_Name}>d__{func_ID}
I need to be sure of this so I can compare the names

**Reply by 333fred** [reply, marked as answer]
This is an implementation detail. Is it likely to change tomorrow? Maybe not. Is it likely to change in the future? Practically guaranteed, particularly with the runtime async work that will happen in .NET 10. What are you actually trying to do?

**Reply by StanLipman** [reply]
I am trying to find the connection between async functions in the profiler, but the statemachine breaks it up
